### ISSUE TITLE
Verified Balance and total Balance not updated after sending

### ISSUE THREAD

**Issue by pacu** [zcash/ZcashLightClientKit#128]
Credits to @ToposWare

Demo App is not reflecting available funds after sending and causes misleading errors in subsequent submissions could be attempted without unspent notes yet available.

![image](https://user-images.githubusercontent.com/484008/82449162-2b87cc00-9a81-11ea-8111-0e567607a623.png)

**Comment by sebastiendan** [comment]
@pacu 
Executing [this SQL request](https://github.com/str4d/librustzcash/blob/note-spending-v7/zcash_client_sqlite/src/transact.rs#L131) on my `ZcashSdk_testnet_data.db` I get no result.

I can provide that db file if that can help.

**Comment by pacu** [comment]
@str4d, can you take a look?

**Comment by pacu** [comment]
bringing some context over from discord conversations:

> This balance of 4.0 ZEC is most likely the result of multiple trials to get ZEC from the testnet faucet. At the time I was going through my first trials (last week), I was using a fork of testnet so didn't receive the tx/coins. After I notified you to get lightwalletd.testnet.z.cash updated I started using a local instance of lightwalletd plugged to a local full node on testnet. Now that I can confirm I'm using the right branch of your testnet (I get the right latestblock), I have the right balance containing the amount coming from the faucet.
What I'm saying with this is that these coins have probably been delivered to my address last week.

**Comment by pacu** [comment]
@sebastiendan once you fixed the lightwalletd issue did you sync from scratch?

**Comment by sebastiendan** [comment]
@pacu Yes, I did (meaning I at least once saw the sync happening from 0 on the sync screen). 

Is there a way on the iOS side to force a re-sync from scratch? Should I delete manually the virtual device data on my disk? (I'm new to iOS dev)

**Comment by pacu** [comment]
the sync always starts from 0, we are not compensating the UI percentage. 
tap the trash can icon on the main screen and restart the app to start over from ground zero

**Comment by sebastiendan** [comment]
I used the trash icon and synced against from zero.

I have now a different error:

<img width="375" alt="Screen Shot 2020-05-22 at 7 55 09" src="https://user-images.githubusercontent.com/12198372/82614052-d9ba7f80-9c01-11ea-97c4-9dc52c1dbbde.png">

**Comment by pacu** [comment]
oh, so, now it's getting to your lwd instance but it's being rejected. that's interesting.

**Comment by sebastiendan** [comment]
My lightwalletd instance outputs only this about the tx:

`
INFO[110457] method called                                 duration=4.6556ms error="<nil>" method=/cash.z.wallet.sdk.rpc.CompactTxStreamer/SendTransaction peer_addr="127.0.0.1:52385"
`

**Comment by sebastiendan** [comment]
Restarting my lwd instance with `--log-level 7` I got this warning: `{"app":"lightwalletd","level":"warning","msg":"CORRUPTION detected in db blocks-cache files, height 0 redownloading","time":"2020-05-22T09:18:52+09:00"}`.

Let me try again sending a tx after lwd has completed the redownload.

**Comment by sebastiendan** [comment]
Same error after a redownload of blocks by lwd and a fresh sync of the iOS app :/

**Comment by sebastiendan** [comment]
About lwd, I got no other logs than before and restarting it shows the CORRUPTION warning again (and lwd redownloads the blocks again)

**Comment by defuse** [comment]
The "bad-txns-sapling-spend-description-invalid" error is because we forgot to update the consensus branch ID prior to Heartwood activation on testnet, so currently the transactions it generates aren't valid on testnet, sorry about that! 

The `CORRUPTION detected in db blocks-cache files` one is unexpected, thanks for reporting!

**Comment by defuse** [comment]
Can you please copy/paste your entire lightwalletd log into a comment here? It'll help us investigate that corruption error. Thanks!

**Comment by sebastiendan** [comment]
@defuse Ok thanks for the update. Any ETA for the consensus branch ID update on testnet?

**Comment by sebastiendan** [comment]
@defuse 

```
sebastiendan@Sebastiens-MBP lightwalletd % ./lightwalletd --no-tls-very-insecure --zcash-conf-path ~/Library/Application\ Support/Zcash/zcash.conf --data-dir . --log-file /dev/stdout --log-level 7
{"app":"lightwalletd","buildDate":"2020-05-19","buildUser":"sebastiendan","gitCommit":"1b0a2d8d57e7bd7665875fdbc950d0697a44d4be","level":"info","msg":"Starting gRPC server version v0.3.0-49-g1b0a2d8 on 127.0.0.1:9067","time":"2020-05-22T10:18:03+09:00"}
{"app":"lightwalletd","level":"warning","msg":"Starting insecure server","time":"2020-05-22T10:18:03+09:00"}
Starting insecure server
{"app":"lightwalletd","level":"info","msg":"Got sapling height 0 block height 925598 chain test branchID 76b809bb","time":"2020-05-22T10:18:03+09:00"}
{"app":"lightwalletd","level":"warning","msg":"lengths file has impossible value 74","time":"2020-05-22T10:18:03+09:00"}
{"app":"lightwalletd","level":"warning","msg":"CORRUPTION detected in db blocks-cache files, height 0 redownloading","time":"2020-05-22T10:18:03+09:00"}
{"app":"lightwalletd","level":"info","msg":"Found 0 blocks in cache","time":"2020-05-22T10:18:03+09:00"}
{"app":"lightwalletd","level":"info","msg":"Ingestor adding block to cache: 2122","time":"2020-05-22T10:18:07+09:00"}
{"app":"lightwalletd","level":"info","msg":"Ingestor adding block to cache: 4193","time":"2020-05-22T10:18:11+09:00"}
```

**Comment by defuse** [comment]
Thanks! My guess would be sometime next week, I'll @ you in this issue once it's fixed.

**Comment by sebastiendan** [comment]
@defuse Awesome, thanks

**Comment by LarryRuane** [comment]
@sebastiendan (@defuse), please try the latest version of master, commit b47c0dd or later (sorry, we'll get better versioning soon), that will fix the corruption problem (see https://github.com/zcash/lightwalletd/pull/268 for details).

**Comment by sebastiendan** [comment]
@defuse Any update on how this is going?

**Comment by defuse** [comment]
It still hasn't been fixed, sorry :(
